aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -1530,6 +1537,28 @@ def __init__(
         self.context: dict = {}  # early init as customers might add context before event resolution
         self.processed_stack_frames = []
         self._response_builder_class = ResponseBuilder[BaseProxyEvent]
+        self._has_response_validation_error = response_validation_error_http_status is not None
+
+        if response_validation_error_http_status and not enable_validation:
+            msg = "'response_validation_error_http_status' cannot be set when enable_validation is False."
+            raise ValueError(msg)
+
+        if (
+            not isinstance(response_validation_error_http_status, HTTPStatus)
+            and response_validation_error_http_status is not None
+        ):
+
+            try:
+                response_validation_error_http_status = HTTPStatus(response_validation_error_http_status)
+            except ValueError:
+                msg = f"'{response_validation_error_http_status}' must be an integer representing an HTTP status code."
+                raise ValueError(msg) from None
+
+        self._response_validation_error_http_status = (
+            response_validation_error_http_status
+            if response_validation_error_http_status
+            else HTTPStatus.UNPROCESSABLE_ENTITY
+        )
```

Comment:
```suggestion
        self._response_validation_error_http_status = self._validate_response_validation_error_status(
            response_validation_error_http_status, 
            enable_validation
        )
```

And then we can simplify the logic and move the code for a specific function:

```python
def _validate_response_validation_error_status(self, response_validation_error_http_status, enable_validation):
	        if response_validation_error_http_status and not enable_validation:
	            raise ValueError("'response_validation_error_http_status' cannot be set when enable_validation is False.")
	
	        if response_validation_error_http_status is None:
	            return HTTPStatus.UNPROCESSABLE_ENTITY
	
	        if isinstance(response_validation_error_http_status, HTTPStatus):
	            return response_validation_error_http_status
	
	        try:
	            return HTTPStatus(response_validation_error_http_status)
	        except ValueError:
	            raise ValueError(f"'{response_validation_error_http_status}' must be an integer representing an HTTP status code.")
```

Reply:
Great idea